src/pages/token/TokenPage.vue

```diff
@@ -0,0 +1,147 @@
+                exact
+                push
+                label="Transfers"
+            />
+            <q-route-tab
+                name="holders"
+                to="#holders"
+                exact
+                push
+                label="Holders"
+            />
+        </q-tabs>
+
+        <q-tab-panels
+            v-model="selectedTab"
+            animated
+            keep-alive
+        >
+            <q-tab-panel name="transfers">
+                <token-transfers
+                    :address="address"
+                    @token-info-loaded="handleTokenInfoLoaded"
+                />
+            </q-tab-panel>
+            <q-tab-panel name="holders" class="shadow-2">
+                <token-holders
+                    :address="address"
+                    @token-info-loaded="handleTokenInfoLoaded"
+                />
+            </q-tab-panel>
+        </q-tab-panels>
+    </div>
+</div>
+</template>
+
+<script>
+import TokenHolders from 'pages/token/TokenHolders';
+import TokenTransfers from 'pages/token/TokenTransfers';
+
+import { toChecksumAddress } from 'src/lib/utils';
+
+const genericLogo = require('assets/telos-logo--evm-generic.png');
+
+const tabs = {
+    holders: 'holders',
+    transfers: 'transfers',
+};
+
+export default {
+    name: 'TokenPage',
+    components: {
+        TokenHolders,
+        TokenTransfers,
+    },
+    data: () => ({
+        tabs,
+        selectedTab: tabs.transfers,
+        tokenInfo: null,
+    }),
+    computed: {
+        address() {
+            return toChecksumAddress(this.$route.params?.address);
+        },
+        headerTokenText() {
+            if (!this.tokenInfo) {
+                return 'Loading...';
+            }
+
+            const { name, symbol } = this.tokenInfo;
+
+            if (name && symbol) {
+                return `${name} (${symbol})`;
+            }
+
+            return 'Unknown Token';
+        },
+        headerTokenLogo() {
+            return this.tokenInfo?.logoURI ?? genericLogo;
+        },
+    },
+    watch: {
+        '$route.hash': {
+            handler(newHash) {
+                const tabNames = Object.values(this.tabs);
+                const hash = newHash?.replace('#', '') ?? '';
+
+                if (!tabNames.includes(hash)) {
+                    this.$router.replace({ hash: this.tabs.transfers });
+                }
+            },
+            immediate: true,
+        },
+        '$route.path': {
+            handler(newPath) {
+                const urlPath = newPath.replace('/token/', '');
+                const urlPathChecksum = toChecksumAddress(urlPath);
+                const pathNeedsFormatting =
+                    !!urlPathChecksum &&
+                    urlPath !== urlPathChecksum &&
+                    urlPath.toLowerCase() === urlPathChecksum.toLowerCase();
+
+                if (pathNeedsFormatting) {
+                    this.$router.replace(`/token/${this.address}`);
+                }
```

Comment:
ensures that all checks against the token contract address will be the correct casing